File: DevExpress-Demo/CS/SchedulingDemo.Wpf/Common/Converters.cs
using DevExpress.Xpf.Scheduling;
using SchedulingDemo.ViewModels;
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Windows;
using System.Windows.Data;
using System.Windows.Markup;

namespace SchedulingDemo {
    public class ResourceNameToIdConverter : MarkupExtension, IValueConverter {
        IList<string> resources;

        public ResourceNameToIdConverter() {
            resources = WorkData.Calendars.Select(x => x.Name).ToList();            
        }

        public override object ProvideValue(IServiceProvider serviceProvider) { return this; }

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            IEnumerable<long> ids = value as IEnumerable<long>;
            if(ids != null) {
                List<string> names = ids.Select(x => resources[(int)x]).ToList();
                return names;
            }
            return new List<string>() { TimeRegionsDemoViewModel.AnyResource };
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {            
            List<object> names = value as List<object>;
            if (names != null && (string)names[0] != TimeRegionsDemoViewModel.AnyResource) {
                IEnumerable<long> ids = names.Select(x => WorkData.Calendars.Where(v => v.Name == (string)x).FirstOrDefault().Id);
                return ids;
            }

            return null;
        }
    }
}

File: DevExpress-Demo/VB/SchedulingDemo.Wpf/Common/Converters.vb
Imports SchedulingDemo.ViewModels
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Linq
Imports System.Windows.Data
Imports System.Windows.Markup

Namespace SchedulingDemo

    Public Class ResourceNameToIdConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Private resources As IList(Of String)

        Public Sub New()
            resources = WorkData.Calendars.[Select](Function(x) x.Name).ToList()
        End Sub

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim ids As IEnumerable(Of Long) = TryCast(value, IEnumerable(Of Long))
            If ids IsNot Nothing Then
                Dim names As List(Of String) = ids.[Select](Function(x) resources(CInt(x))).ToList()
                Return names
            End If

            Return New List(Of String)() From {TimeRegionsDemoViewModel.AnyResource}
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Dim names As List(Of Object) = TryCast(value, List(Of Object))
            If names IsNot Nothing AndAlso Not Equals(CStr(names(0)), TimeRegionsDemoViewModel.AnyResource) Then
                Dim ids As IEnumerable(Of Long) = names.[Select](Function(x) Enumerable.FirstOrDefault(Enumerable.Where(WorkData.Calendars, Function(v) Equals(v.Name, CStr(x)))).Id)
                Return ids
            End If

            Return Nothing
        End Function
    End Class
End Namespace
